domain/access/state/permission.go

```diff
@@ -329,38 +333,80 @@ AND     u.removed = false
 // exist, it is created. An accesserrors.AccessNotFound error is returned if no
 // access can be found for this user, and (only in the case of external users),
 // the everyone@external user.
-func (st *PermissionState) ReadUserAccessLevelForTargetAddingMissingUser(ctx context.Context, subject user.Name, target corepermission.ID) (corepermission.Access, error) {
-	userAccess, err := st.ReadUserAccessLevelForTarget(ctx, subject, target)
-	if err == nil {
-		return userAccess, nil
-	} else if !(errors.Is(err, accesserrors.AccessNotFound) && !subject.IsLocal()) {
-		// If there is an access not found error and the user is external,
-		// continue. Otherwise, return the error.
-		return userAccess, errors.Trace(err)
-	}
+func (st *PermissionState) ReadUserAccessLevelForTargetAddingMissingUser(
```

Comment:
question: Why are we creating a user on a read method? If you create the user, you can't add the access.

Reply:
This was part of the compromise we can up for for external user login. It checks the login and creates the external user if they are not known. 

I would like to rename this method TBH, looking at it now its not a good name.

Reply:
Rename it and well comment it.

Reply:
Ok, will do it in #17955 because those changes are cherry-picked from that PR.